### PR TITLE
Add option to disable homing safety checks

### PR DESCRIPTION
**Pull request by towe96**
module: Allow users to disable homing safety checks

Adds an option under the [force_move] configuration section to set ignore_homing to True.

This disables the endstop checking in the kinematics file for Cartesian, CoreXY and CoreXZ printers, allowing all manual moves.

If enabled on a non-support kinematic (e.g. Delta), an error is raised.

This attempts to solve issue #335 

Signed-off-by: Tobias Weiß <t.weiss@bk.ru>

### CODE REVIEW

File: klippy/kinematics/cartesian.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def __init__(self, toolhead, config):
                                               above=0., maxval=max_velocity)
         self.max_z_accel = config.getfloat('max_z_accel', max_accel,
                                            above=0., maxval=max_accel)
+        self.ignore_homing = config.getsection('force_move').getboolean(
```
Comment (FHeilmann) [inline klippy/kinematics/cartesian.py:30]: what happens here when there is no section `force_move` in the config? I think the way this is implemented currently, klipper will error out if no such section is present.

Comment (towe96) [reply klippy/kinematics/cartesian.py:30]: I tried that, it assumes the False value from getboolean()

Comment (FHeilmann) [reply klippy/kinematics/cartesian.py:30]: I see, thanks for the reply!

**Comment by KevinOConnor** [comment]
Thanks, but I my high-level feedback is that I'm not sure this is a good candidate for the main Klipper repository.  There is already a way to disable the homing checks (by issuing a SET_KINEMATIC_POSITION command), and I'd prefer not to add additional complexity to the low-level kinematic classes to check for a rare option.

-Kevin

**Comment by towe96** [comment]
I don't think this is comparable with `SET_KINEMATIC_POSITION`s everywhere - like I'd been using previously in my start g-code as well as some macros to work around this issue.
I feel like Klipper does need a way of easily disabling this safety feature, like in Marlin where it's not even on by default (except for Deltas, which this doesn't touch).

It might be cleaner to put the check into `_check_endstops` though, since it'd only need to appear there once instead of being applied for both X/Y and Z. Would you prefer that?

**Comment by KevinOConnor** [comment]
I'm going to close this as I don't think it is a good choice for the master Klipper branch.

-Kevin

**Comment by towe96** [comment]
How else can we fix this essential lack of a feature?

**Comment by vicenzajay** [comment]
Bumping this.  I just installed Klipper and am tuning resonance now, but this is a dealbreaker.  There are times, especially during calibration, when you need to stop a print.  Without being able to manually raise the z axis of the printhead, disentangling the head from the print itself is problematic (and, in some cases, conducive to damage).  There really has to be a way of manually moving the printhead and/or bed without homing.  This is a no-brainer.

**Comment by vicenzajay** [comment]
If there's a script out there to add to printer.cfg to allow this, that would work as well, but not having control over my axes' positioning in a manual fashion just isn't an option.  It makes Klipper less flexible than Marlin in one significant aspect - and that aspect deals with user control of the printer.  This is part of the appeal of Klipper, and why it's an enthusiast firmware.  I've read the threads, and I "somewhat" understand the reasoning; however, there should be someway of implementing the DO_YOU_REALLY_WANT_TO_DO_THIS option.

**Comment by Shonky-Donkey** [comment]
Just installed Klipper to discover this very frustrating "feature"

A manual button to override (say, clicking the "motors off" button in Fluidd to force them on seems like it would be enough of a check to make sure people intended to do this. Maybe with a warning or something.

Kevin, just so you realize... it's frustrating enough to me that I signed up for a github account just to write this. As more people use Klipper, more people are going to want this.

**Comment by FHeilmann** [comment]
> Just installed Klipper to discover this very frustrating "feature"
> 
> A manual button to override like clicking the "motors off" button seems like it would be enough of a check to make sure people intended to do this. Maybe with a warning or something.
> 
> Kevin, just so you realize... it's frustrating enough to me that I signed up for a github account just to write this. As more people use Klipper, more people are going to want this.

```
[force_move]
enable_force_move: True

[gcode_macro UNLOCK_MOTORS]
gcode:
  SET_KINEMATIC_POSITION X=0 Y=0 Z=0
```

add this to your config, and, assuming you're using mainsail or fluidd, you'll get your button in the webinterface.

**Comment by Shonky-Donkey** [comment]
perfect, works like a charm, thanks FHeilmann. Also is a good demonstration of Klipper's macro functionality. 

I really like the config file in Klipper vs recompiling to make changes in Marlin.

**Comment by vicenzajay** [comment]
Thanks for the macro....that will help!

**Comment by towe96** [comment]
``You can also give no values to `SET_KINEMATIC_POSITION`, useful if you've just unlocked the motors but didn't move the carriage.

Another options is manually removing the `self._check_endstops(move)` lines from `klippy/kinematics/cartesian.py`, which will disable the homing checks entirely. You may need to do this again after `git pull`s.

If you don't want to modify your "low level" code but still kill all the movement checks, this is what I'm currently using for macros:
```
[force_move]
enable_force_move: True

[gcode_macro M18]
gcode: SET_STEPPER_ENABLE STEPPER=extruder ENABLE=0
       SET_STEPPER_ENABLE STEPPER=stepper_x ENABLE=0
       SET_STEPPER_ENABLE STEPPER=stepper_y ENABLE=0
       SET_STEPPER_ENABLE STEPPER=stepper_z ENABLE=0
       SET_KINEMATIC_POSITION
rename_existing: M181

[gcode_macro M84]
gcode: M18
rename_existing: M841

[delayed_gcode FixMovementBug]
gcode: M18
initial_duration: 0.1
```
This redefines M18 to disable the steppers without enabling the homing check, and running M18 0.1 seconds after startup ensures the check is disabled there as well.
Again, as FHeilmann noted, you'll probably want to give `SET_KINEMATIC_POSITION` coordinates where you carriage can move in any direction.

**Pull request closed without merging**